lib/index.js

```diff
@@ -300,73 +299,69 @@ internals.stop = async (server) => {
 internals.state = (realm) => Toys.state(realm, 'schwifty');
 internals.rootState = (realm) => Toys.rootState(realm, 'schwifty');
 
-internals.migrateDiffCmd = async (srv, args, rootDir, ctx) => {
+internals.migrateDiffCmd = async (srv, argv, rootDir, ctx) => {
 
-    const SchwiftyMigrateDiff = SchwiftyMigrateDiffHelper.get();
+    const { DisplayError, colors, options } = ctx;
+    const migrateDiff = Helpers.getSchwiftyMigrateDiff();
 
-    if (!SchwiftyMigrateDiff) {
-        throw new ctx.DisplayError('To use this command please first install `npm install --save-dev schwifty-migrate-diff`');
+    if (!migrateDiff) {
+        throw new DisplayError(colors.yellow('To use this command please first install `npm install --save-dev schwifty-migrate-diff`'));
     }
 
+    // Parse command line args
+
     const definition = {
         mode: {
-            description: 'Set mode to "create" or "alter" (defaults to "alter") -- when set to create the migration file ignores changes that will delete or mutate existing columns/tables'
-        },
-        dir: {
-            description: 'Specifies the directory in which to write the migration file, defaults to the migrationsDir Schwifty receives at registration'
+            type: 'string',
+            description: 'Determines the types of changes included in the migration. The default "alter" includes both non-destructive and destructive changes (e.g. altering a column type), while "create" only includes non-destructive changes (e.g. adding a new column).',
+            valid: ['alter', 'create'],
+            default: 'alter'
         }
     };
 
-    const parsedArgs = Bossy.parse(definition, { argv: args });
+    const parsedArgs = Bossy.parse(definition, { argv });
+    const usage = () => Bossy.usage(definition, 'hpal run schwifty:migrate:diff <migration-name> [--mode alter|create] [--dir <migrations-dir>]', { colors: options.colors });
 
     if (parsedArgs instanceof Error) {
-        throw new ctx.DisplayError(parsedArgs.message + Os.EOL + Bossy.usage(definition));
+        throw new DisplayError(`${usage()}\n\n${colors.red(parsedArgs.message)}`);
     }
 
-    Hoek.assert(parsedArgs._ && parsedArgs._.length !== 0, new ctx.DisplayError('A name must be specified for the generated migration'));
-    Hoek.assert(parsedArgs._.length === 1, new ctx.DisplayError(Bossy.usage(definition)));
+    const [migrationName, ...others] = parsedArgs._ || [];
 
-    const [migrationName] = parsedArgs._;
+    Hoek.assert(migrationName, new DisplayError(`${usage()}\n\n` + colors.red('You must specify a <migration-name>.')));
+    Hoek.assert(!others.length, new DisplayError(`${usage()}\n\n` + colors.red('Did you try to specify something other than a <migrations-dir> or <mode>?')));
```

Comment:
Do we care about using Os.EOL here?

Reply:
`EOL` is primarily for files, as I think about it.  You'll find that `console.log()`, for example, just uses `\n` in CLI output.

Reply:
https://github.com/nodejs/node/blob/master/lib/console.js#L141-L159

Reply:
=OOOO
Wooahh I thought `\n` was Mac specific for a long time then! Oh boy loLL